isce2_topsapp/localize_burst.py

```diff
@@ -0,0 +1,315 @@
+
+    @staticmethod
+    def reformat_gcp(point):
+        attribs = ['line', 'pixel', 'latitude', 'longitude', 'height']
+        values = {}
+        for attrib in attribs:
+            values[attrib] = float(point.find(attrib).text)
+        return values
+
+    def create_gcp_df(self):
+        points = self.annotation.findall('.//{*}geolocationGridPoint')
+        gcp_df = pd.DataFrame([self.reformat_gcp(x) for x in points])
+        gcp_df = gcp_df.sort_values(['line', 'pixel']).reset_index(drop=True)
+        return gcp_df
+
+    def create_geometry(self, gcp_df):
+        burst_index = self.burst_number - 1
+        lines = int(self.annotation.findtext('.//{*}linesPerBurst'))
+        first_line = gcp_df.loc[gcp_df['line'] == burst_index * lines, ['longitude', 'latitude']]
+        second_line = gcp_df.loc[gcp_df['line'] == (burst_index + 1) * lines, ['longitude', 'latitude']]
+        x1 = first_line['longitude'].tolist()
+        y1 = first_line['latitude'].tolist()
+        x2 = second_line['longitude'].tolist()
+        y2 = second_line['latitude'].tolist()
+        x2.reverse()
+        y2.reverse()
+        x = x1 + x2
+        y = y1 + y2
+        footprint = geometry.Polygon(zip(x, y))
+        centroid = tuple([x[0] for x in footprint.centroid.xy])
+        return footprint, footprint.bounds, centroid
+
+
+def create_burst_request(burst_params: BurstParams, content: str) -> dict:
+    urls = {
+        'metadata': 'https://g6rmelgj3m.execute-api.us-west-2.amazonaws.com/metadata',
+        'geotiff': 'https://g6rmelgj3m.execute-api.us-west-2.amazonaws.com/geotiff',
+    }
+    payload = {
+        'zip_url': burst_params.safe_url,
+        'image_number': str(burst_params.image_number),
+        'burst_number': str(burst_params.burst_number),
+    }
+    return {
+        'url': urls[content],
+        'params': payload,
+    }
+
+
+def download_metadata(
+        asf_session: requests.Session,
+        burst_params: BurstParams,
+        out_file: Union[Path, str] = None) -> ET.Element:
+    burst_request = create_burst_request(burst_params, content='metadata')
+    burst_request['cookies'] = {'asf-urs':asf_session.cookies['asf-urs']}
+
+    response = asf_session.get(**burst_request)
+    response.raise_for_status()
+
+    metadata = ET.fromstring(response.content)
+
+    if out_file:
+        ET.ElementTree(metadata).write(out_file, encoding='UTF-8', xml_declaration=True)
+
+    return metadata
+
+
+def download_geotiff(
+        asf_session: requests.Session,
+        burst_params: BurstParams,
+        out_file: Union[Path, str]) -> str:
+    burst_request = create_burst_request(burst_params, content='metadata')
+    burst_request['cookies'] = {'asf-urs':asf_session.cookies['asf-urs']}
+
+    i = 1
+    downloaded = False
+    while (not downloaded) and (i <= 3):
+        print(f'Download attempt #{i}')
+        response = asf_session.get(**burst_request)
+        downloaded = response.ok
+        i += 1
```

Comment:
```suggestion
    for ii in range(1, 4):
        print(f'Download attempt #{ii}')
        response = asf_session.get(**burst_request)
        if (downloaded := response.ok):
            break
```
while loop with an iterator is just a for loop.